src/flb_file.c

```diff
@@ -49,7 +49,7 @@ flb_sds_t flb_file_read(const char *path)
         goto err;
     }
 
-    result = flb_sds_create_size(flen);
+    result = flb_sds_create_size(flen + 1);
```

Comment:
sds always set automatically an extra byte with a NULL value, so this extra byte is not required

Reply:
I will adjust in #5269, where I've extracted this change to.